### ISSUE TITLE
Tempo: The contents of the Name column in search results can be a little confusing

### ISSUE THREAD

**Issue by mdisibio** [grafana/grafana#70932]
### 
# What went wrong?

**What happened**:

Tempo TraceQL search results can have a Name column with values from both traces and spans.  They look similar at first but they are a little different and this can sometimes be confusing.  The trace-level value is a combination of both the service name and span name, while the span-level value is just the span name.  Perhaps there are some ways to improve the readability here. 

<img width="1102" alt="image" src="https://github.com/grafana/grafana/assets/13524475/a64d05a2-94b0-4d7f-b518-81258f16431e">

### 
# How do we reproduce it?

Step 1:  Perform a search in TraceQL that uses the span name. This causes the span name to be shown in the results, in the same Name column as the trace-level data.
     i.e. { name = "foo" }

### 
# What Grafana version are you using?

10.0

### 
### Grafana Platform?

None

### Is this a Regression?

None

**Comment by 09jvilla** [comment]
Note I think this might be a duplicate of this related issue:
https://github.com/grafana/grafana/issues/64840

**Comment by 09jvilla** [comment]
heads up @cyrille-leclerc

**Comment by cyrille-leclerc** [comment]
> The simplest thing to do seems like it would be to take the `Name` column at the trace level and split it into `name` and `service.name` since [both are always returned in the response](https://github.com/grafana/grafana/issues/64840#issuecomment-1470481555).

I would see

| Date               | Service       | Name                             | Duration | Trace ID |
|--------------------|---------------|----------------------------------|----------|----------|
| 2023-06-29 19:32:52| tempo-gateway | GET /tempo/api/traces/${traceid} | 221ms  | 562df44eeea3d7c6 |

With:
* Date: timestamp of the root span
* Service: `service.name` of the root span
* Name: `span.name` of the root span
* Duration: duration of the root span
* Trace ID: the id of the trace

I would also make the table row selectable/clickable rather than having to click on one specific cell.

**Comment by cyrille-leclerc** [comment]
Thanks Joey, the click ability of the row is a minor detail in comparison to the other topics of this conversation.
I would be in favor of descoping this click ability.

**Comment by cyrille-leclerc** [comment]
Great catch @09jvilla, I'm sorry I missed this point .

As App O11y is controlling the TraceQL requests in the App O11 Service Traces tab, I would imagine that App O11y would enrich its TraceQL template the same way it enriches its LogQL query today.

By the way  but it may be a subsequent conversation: as TraceQL adds support for `select`, will the "Trace Search Result table" adapt its column to the selected attributes?

I could imagine as SQL style query like:

```
select startTime as Date, service.name as Service, name as Name, duration as Duration, traceId as "Trace ID"...
```

**Comment by 09jvilla** [comment]
> Great catch @09jvilla, I'm sorry I missed this point .

Np!

> As App O11y is controlling the TraceQL requests in the App O11 Service Traces tab, I would imagine that App O11y would enrich its TraceQL template the same way it enriches its LogQL query today.

Great - I was hoping you'd say that. =) I imagine there are more experiences in app o11y where you don't even need to show the raw TraceQL and the expectation is that at the point that the user needs to make a significant change they'd be kicked from the app o11y app into the Explore anyways. 

> By the way but it may be a subsequent conversation: as TraceQL adds support for select, will the "Trace Search Result table" adapt its column to the selected attributes?

Joey to confirm, but I would imagine so. After all, it does adapt the columns today based on which fields you are filtering on. Today you see one column for every field your query filters on, so the columns displayed vary by query. With the select capability, I would now expect to see a column for every field my query filters on and every field that I've selected. Arguably, if you wanted to be even more SQL-ish, you could consider no longer returning columns for the fields being filtered on, and just having columns for the fields that are selected (similar to how in SQL, you only see the fields that you explicitly select; if not selected, you wouldn't see the fields that you filtered by in a `where` clause).

**Comment by cyrille-leclerc** [comment]
Great alignment and split of roles & responsibilities to move fast!

> I imagine there are more experiences in app o11y where you don't even need to show the raw TraceQL and the expectation is that at the point that the user needs to make a significant change they'd be kicked from the app o11y app into the Explore anyways.

@bfmatei is already brainstorming on how to hide by default the verbose LogQL in the logs view 😄

**Comment by joey-grafana** [comment]
The way in which the select functionality was described does make sense, however it would need some more discussion IMO before we can be sure.

**Comment by 09jvilla** [comment]
> The way in which the select functionality was described does make sense, however it would need some more discussion IMO before we can be sure.

Agreed. 
That being said, I don't think `select` is necessary to just do the smaller scope change proposed [here](https://github.com/grafana/grafana/issues/70932#issuecomment-1642209888), which I still think would address the initial purpose of this issue. 
Thoughts on just focusing on that change for now and then opening a subsequent issue from there?

**Comment by 09jvilla** [comment]
@joey-grafana -- Is this the right PR that would resolve this?
https://github.com/grafana/grafana/pull/72449

**Comment by joey-grafana** [comment]
It is indeed :) hope to be merging soon

**Comment by g3david405** [comment]
Can we customize the trace name?

**Comment by g3david405** [comment]
For differenent dotnet version, some old version do not have http.route attribute.
So we want to use url.path instead.

**Comment by joey-grafana** [comment]
This is set to the span name / operation name. It is not currently customisable.

**Comment by g3david405** [comment]
@joey-grafana could you provide location of the source code?
Maybe I can change it and build a image for me.
Thanks

**Comment by MrMegaMango** [comment]
Does it only show tracer_provider resource service name and span_name not the tracer_name in `tracer_provider.get_tracer`?
